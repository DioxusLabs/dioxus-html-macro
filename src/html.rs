use crate::prelude::*;

#[derive(Default)]
pub struct Html {
    pub elements: Vec<Item>,
}

impl Parse for Html {
    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
        let mut elements = vec![];
        while !input.is_empty() && !input.peek2(Token![/]) {
            elements.push(input.parse()?);
        }
        Ok(Html { elements })
    }
}

impl ToTokens for Html {
    fn to_tokens(&self, tokens: &mut TokenStream) {
        for element in &self.elements {
            tokens.extend(quote! {
                #element
            });
        }
    }
}
