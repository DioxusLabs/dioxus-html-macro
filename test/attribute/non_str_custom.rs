use dioxus_html_macro::html;

fn main() {
    
    html!(
        <div "custom"={move |_| ()}>
        </div>
    );
}
