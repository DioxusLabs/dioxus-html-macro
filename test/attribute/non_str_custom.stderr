error: expected a string literal.
 --> test/attribute/non_str_custom.rs:6:24
  |
6 |         <div "custom"={move |_| ()}>
  |                        ^^^^
